### PR TITLE
Add Levenberg-Marquardt algorithm and modify curve fitting method

### PR DESCRIPTION
**Pull request by Gene0315**
This PR
1. adds Levenberg-Marquardt algorithm
2. revises curve fitting method
3. splits document into chapters
4. adjusts the line length to be consistent with pdf

### CODE REVIEW

**Review by yungyuc — CHANGES_REQUESTED** [review]
Could you please keep all notes in one (original) file?

**Comment by Gene0315** [comment]
> Could you please keep all notes in one (original) file?

@yungyuc It's ok. I have already done it. But, can I ask for the reason?
By the way, I still keep the line length (120 characters) consistent with pdf. If you'd like it restored to 74 characters per line, please let me know.

File: foil.tex [hunk 1]
```diff
@@ -8,56 +7,40 @@
 % chktex-file 25
 % chktex-file 36
 % chktex-file 37
+% (C) 2024 Yung-Yu Chen.  All rights reserved.
```
Comment (yungyuc) [inline foil.tex]: Why do you change the position of the copyright line?  It's usually at the beginning of the file.

But you should add your name it it.

File: foil.tex [hunk 2]
```diff
@@ -8,56 +7,40 @@
...
-}
-
-\author{
-%
-Jie-Yin Lin and Yung-Yu Chen
-%
-}
-
-%\date{2008.6.4}
+\title{Fit Four-digit NACA Airfoil with Cubic Bezier Curve}
```
Comment (yungyuc) [inline foil.tex]: Please keep the title and author formatting.

**Review by yungyuc — CHANGES_REQUESTED** [review]
> > Could you please keep all notes in one (original) file?
> 
> @yungyuc It's ok. I have already done it. But, can I ask for the reason?

A single file makes it easier to edit.  Multiple files make it hard for _everyone_ in a team to track what to change.  It is usually productive to edit a file not exceeding 20k lines.  

> By the way, I still keep the line length (120 characters) consistent with pdf. If you'd like it restored to 74 characters per line, please let me know.

It's OK to change, but not in this PR.  Change of source text formatting is refactoring, and refactoring should always use standalone PRs.

---

- [x] Please revert change of line width from this PR.  Keep only change of contents.  It is OK to change "code" (text) style, but it is considered refactoring.
- [x] [`L21`](https://github.com/yungyuc/bezierfoil/pull/14/files#diff-b89d7d2901eef4448199c8d7decf665ab81b21f4de95d81d4a83cb50972a78a8R21): Keep title and author formatting in source.

**Comment by Gene0315** [comment]
> * [ ]  Please revert change of line width from this PR.  Keep only change of contents.  It is OK to change "code" (text) style, but it is considered refactoring.
> * [ ]  [`L21`](https://github.com/yungyuc/bezierfoil/pull/14/files#diff-b89d7d2901eef4448199c8d7decf665ab81b21f4de95d81d4a83cb50972a78a8R21): Keep title and author formatting in source.

What I do in this [commit](https://github.com/yungyuc/bezierfoil/pull/14/commits/2309af03ed6eabbc661ed7baa0c2bc7b700f8c85) :
* [ ] revert change of line width. 
* [ ] recover the original title and author formatting.
* [ ] adjust some line width to make it easier to compare the changes. (If I understand correctly, the purpose of this PR is only to see change of contents.)

I'm sorry for causing the inconvenience, and I will open another PR to change the formatting after closing this PR.

File: foil.tex [hunk 3]
```diff
@@ -72,11 +65,11 @@ \chapter{Four-digit NACA Airfoil}
     \item
     Camber line (red dashed line)
     \item
-    Leading edge ($x = 0, y = 0$)
+    Leading edge ($x=0, y=0$)
```
Comment (yungyuc) [inline foil.tex]: Please revert unnecessary change of white spaces, blank lines and other minor formatting.

In a PR adding and changing (editing) contents, formatting changes should be avoided.  On the other hand, when changing formatting/typesetting, do not make content change.  Mixing content and formatting changes will make review hard.

This is a principle derived from computer code.

File: foil.tex [hunk 4]
```diff
@@ -72,11 +65,11 @@ \chapter{Four-digit NACA Airfoil}
...
     \item
-    Trailing edge ($x = c, y = 0$)
+    Trailing edge ($x=c, y=0$)
 \end{enumerate}
-\begin{figure}[h]
+\begin{figure}[h!]
```
Comment (yungyuc) [inline foil.tex:81]: This typesetting change is fine.

File: foil.tex [hunk 5]
```diff
@@ -179,41 +180,156 @@ \chapter{Cubic Bezier Curve}
...
 curves.
 
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%%
+\chapter{Levenberg-Marquardt Algorithm}
+% ref: https://en.wikipedia.org/wiki/Levenberg%E2%80%93Marquardt_algorithm
+%%
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+Levenberg-Marquardt algorithm (LM) is used to solve non-linear least
```
Comment (yungyuc) [inline foil.tex:238]: It's a good summary.  Thanks.

File: foil.tex [hunk 6]
```diff
@@ -326,7 +518,6 @@ \chapter{Fit Symmetry Four-digit NACA Airfoil with Cubic Bezier Curve}
 %\bibliographystyle{myunsrtnat} % no sort (order in appearance)
 \bibliographystyle{myplainnat} % sort by author
 \bibliography{turgon_main}
-
```
Comment (yungyuc) [inline foil.tex]: Unnecessary change of blank lines (new lines).

**Review by yungyuc — CHANGES_REQUESTED** [review]
> What I do in this [commit](https://github.com/yungyuc/bezierfoil/pull/14/commits/2309af03ed6eabbc661ed7baa0c2bc7b700f8c85) :
> 
> * [ ]  revert change of line width.
> * [ ]  recover the original title and author formatting.
> * [ ]  adjust some line width to make it easier to compare the changes. (If I understand correctly, the purpose of this PR is only to see change of contents.)

Correct.

> I'm sorry for causing the inconvenience, and I will open another PR to change the formatting after closing this PR.

It's fine.  And it is good to create an issue to track the formatting change.

You can also use a distinct local branch to keep your formatting change right away.  But formatting changes usually conflict with content changes.  If you do them in parallel, it is likely that you will need to resolve a lot of merge/rebase conflicts.

It is a common workflow to edit the contents and format it in batch after a series of editing.  It is usually more productive.

---

- [x] [`L68`](https://github.com/yungyuc/bezierfoil/pull/14/files#diff-b89d7d2901eef4448199c8d7decf665ab81b21f4de95d81d4a83cb50972a78a8R68): Revert unnecessary formatting change.

**Comment by yungyuc** [comment]
@Gene0315 It now looks better.  Are you ready for merging?

**Comment by Gene0315** [comment]
> @Gene0315 It now looks better. Are you ready for merging?

Yes!

**Review by yungyuc — APPROVED** [review]

**Comment by yungyuc** [comment]
Please make an explicit global comment for what you want for the next step, including code review, merging, etc.

**Pull request merged**